wso2/product-apim-tooling
# Adding Integration tests -[2nd Set of Tests]

## Purpose
Adding Integration tests to apictl

## Goals
Automating Integration tests

## Automation tests
### Covered Automated Tests from this PR
**Tests Related to APIs**

1. Export Apis with export-apis Command 

**Tests Related to Dev-first Approach**

1. Export an API with documentation
2. Export an API with a thumbnail (image)
3. Export an API with a thumbnail (on)
4. Import an API with documentation
5. Import an API with a thumbnail (image)
6. Import an API with a thumbnail (icon)
7. Check documentation updates of an API
8. Check thumbnail image update of an API
9. Check thumbnail icon update of an API
10. Initialize an API the definition.yaml (apictl init SampleAPI --definition definition.yaml)

**Tests Related to Applications**

1. List all the applications in an environment (by specifying the owner)


## Test environment
OS - Ubuntu 20.04 LTS
Java - JDK 1.8_252
APIM - 3.2.0
GO - 1.14
## Learning
> Describe the research phase and any blog posts, patterns, libraries, or add-ons you used to solve the problem.
